### PR TITLE
cockpit management extension

### PR DESCRIPTION
**Pull request by aeronautes**
Here is an extension for cockpit control from within scratchpad. Base functionality includes: 

- one click inputing latlong into aircraft nav system with waypoint sequencing
- full access to all cockpit controls defined in DCS macro system and export APIs available to run via UI buttons or as lua scripts written in scratchpad pages. 
- convenience APIs on top of macro system for ease of programming cockpit actions
- persistant waypoints saved to files as mission plans
- customizations per aircraft for higher order functionality such as one click engine start, mfd page setup, fencing in. All user modifiable and contained within scratchpad mod.
- programmable UI buttons for customizations
- dynamic notifications in titlebar
- help page

Looks like this:
apit

You can ignore the changes to scratchpad-hook.lua here as they are mirrored with PR#60

### CODE REVIEW

File: Scripts/Scratchpad/Extensions/lib/AV8BNA.lua [hunk 1]
```diff
@@ -0,0 +1,55 @@
+ft = {}
```
Comment (rkusa) [inline Scripts/Scratchpad/Extensions/lib/AV8BNA.lua:1]: I've only checked #60 for now, but have an early question. Are the files in the `lib` directory written by you, or sourced from somewhere else? If latter, is there a license for those files that we should put next to them?

Comment (aeronautes) [reply Scripts/Scratchpad/Extensions/lib/AV8BNA.lua:1]: Everything in this PR is original code. The lib\ files are mostly API calls that are created in this PR (tt(), etc). A few of the function calls are DCS Exports API. The reason I put them into a different directory than Disabled\ or Extensions\ is that they are not scratchpad compatible lua files. They rely on APIs from aeronautes-pit.lua. Originally they were in the Scratchpad\ pages directory but that got cluttered quickly. If you have a suggestion for other directory or names, I'd be open to discussing.

Comment (rkusa) [reply Scripts/Scratchpad/Extensions/lib/AV8BNA.lua:1]: Thanks for confirming. Location is fine with me – might make sense to group them into `lib/aeronautes-pit/*`, but I do not have a strong opinion about that.

**Comment by aeronautes** [comment]
I've introduced another extension, aeronautes-msgs, in this PR. It saves the 3 different types of messages a player can receive and displays the history when the associated button is clicked. It will switch the scratchpad buffer to one named, aeronautes-msgs.txt, and display the entire history of that message type for the current session. It will scroll to the bottom, the most recent messages. It will not automatically update the scratchpad display as new messages arrive because of focus and keyboard lock issues. Instead it requires the user to click on the UI button to refresh the buffer. The 3 types of messages are 'chat' which is the in game chat messages with other players displayed on the left side. The 'radio' messages are text renderings of vocal messages such as interaction with AWACS or ATC. These are usually seen in the top left corner of the player screen. And the 'trigger' messages are the other non-verbal messages such as autostart messages and server/mission generated messages. These are normally seen in the top right corner of the screen.
Screen_240702_212715

File: Scripts/Scratchpad/Extensions/lib/AV8BNA.lua (hunk 1 above)
Comment (aeronautes) [reply Scripts/Scratchpad/Extensions/lib/AV8BNA.lua]: renamed directory to aeronautes-lib

**Comment by aeronautes** [comment]
This PR has been revised and replaced with PR 65.

**Pull request closed without merging**